### PR TITLE
Draft of Read the Source: Encryption Draft Bill

### PR DESCRIPTION
**Pull request by ncallaway**
Soliciting feedback on this draft blog post on the proposed encryption bill

### CODE REVIEW

File: _posts/2016-04-24-burr-feinsten-and-encryption.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+> (1) IN GENERAL.—Notwithstanding any other provision of law and except as provided in paragraph (2), a covered entity that receives a court order from a government for information or data shall—
+
+> <p class="indent"> (A) provide such information or data to such government in an intelligible format; or</p>
+
+> <p class="indent">(B) provide such *technical assistance as is necessary* to obtain such information or data in an intelligible format or to achieve the purpose of the court order.</p>
+
+This section says that any _covered entity_ that receives a court order for data must provide the data or must work to produce the data. The phrase in bold — technical assistance as is necessary — is actually very important.
+
+Most court orders to third parties require them to provide "reasonable assistance" to facilitate court order. This proposed law goes a _significant_ step further and requires a covered entity to provide "assistance as necessary". The difference between these two levels of assistance is quite large. [^1]
```
Comment (egreer) [inline _posts/2016-04-24-burr-feinsten-and-encryption.md]: Should this be ^2

File: _posts/2016-04-24-burr-feinsten-and-encryption.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+> (2) SCOPE OF REQUIREMENT.—A covered entity that receives a court order referred to in paragraph (1)(A) shall be responsible only for providing data in an intelligible format if such data has been made unintelligible by a feature, product, or service owned, controlled, created, or provided, by the covered entity or by a third party on behalf of the covered entity.
+
+This section says that you are only required to provide data if _you_ provided the service that encrypted the data.
+
+>(3) COMPENSATION FOR TECHNICAL ASSISTANCE.—A covered entity that receives a court order from a government as described in paragraph (1) and furnishes technical assistance under subparagraph (B) of such paragraph pursuant to such order shall be compensated for such costs as are reasonably necessary and which have been directly incurred in providing such technical assistance or such data in an intelligible format.
+
+This section says that the government will pay you for your time in providing the technical assistance. However, they won't compensate you for any indirect losses from providing the technical assistance. [^2]
```
Comment (egreer) [inline _posts/2016-04-24-burr-feinsten-and-encryption.md]: Should this be ^3?

**Comment by egreer** [comment]
This is really good, also I contacted both of my senators who are on the committee.

File: _posts/2016-04-24-burr-feinsten-and-encryption.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+[^3]: Again, as an example, let's consider a small software company. If customers left your service after learning that you had the technical ability to decrypt their data you would not be compensated for those losses.
+
+[^4]: This is _one_ of the [many requirements](https://www.owasp.org/index.php/Password_Storage_Cheat_Sheet) of storing passwords securely. As always you should leverage the work of well-used and well-audited cryptographic libraries produced by security experts, instead of rolling your own cryptography.
+
+[^5]: [Well, hash functions would be illegal under poorly drafted legislation. http://www.ibtimes.com/feinstein-burr-draft-encryption-bill-would-require-tech-companies-decrypt-messages-2350722 … @SenFeinstein @SenatorBurr #infosec #privacy](https://twitter.com/garrettdfelix/status/718535330991882241) (@garrettdfelix)
```
Comment (HenryKeiter) [inline _posts/2016-04-24-burr-feinsten-and-encryption.md]: This link/retweet monstrosity is pretty hard to look at.

**Comment by wkirby** [comment]
Good post. I've pushed an update that only really affects the way the article was marked up in markdown. The one place where the correct markup doesn't render quite right (in the multi-paragraph footnote), I've commented on an open bug on Redcarpet to hopefully get it fixed.

The only copy change I made was to indicate that emphasis in quoted sections of the bill are from the author.

**Pull request merged**